ext/json/json.c

```diff
@@ -243,11 +284,20 @@ static PHP_FUNCTION(json_encode)
 	php_json_encode_init(&encoder);
 	encoder.max_depth = (int)depth;
 	php_json_encode_zval(&buf, parameter, (int)options, &encoder);
-	JSON_G(error_code) = encoder.error_code;
 
-	if (encoder.error_code != PHP_JSON_ERROR_NONE && !(options & PHP_JSON_PARTIAL_OUTPUT_ON_ERROR)) {
-		smart_str_free(&buf);
-		RETURN_FALSE;
+	if (0 == (options & PHP_JSON_THROW_EXCEPTIONS)) {
+		JSON_G(error_code) = encoder.error_code;
+		if (encoder.error_code != PHP_JSON_ERROR_NONE && !(options & PHP_JSON_PARTIAL_OUTPUT_ON_ERROR)) {
+			smart_str_free(&buf);
+			RETURN_FALSE;
+		}
+	} else {
+		JSON_G(error_code) = PHP_JSON_ERROR_NONE;
+		if (encoder.error_code != PHP_JSON_ERROR_NONE) {
```

Comment:
So this basically means that this has priority over `PHP_JSON_PARTIAL_OUTPUT_ON_ERROR`. Personally I think it should be other way round and it should be ignored if user select `PHP_JSON_PARTIAL_OUTPUT_ON_ERROR`

Reply:
I changed this after thinking about it.